src/components/svg-wrapper.tsx

```diff
@@ -151,6 +185,26 @@ const SvgWrapper = () => {
     const handleBackgroundUp = useEvent((e: React.PointerEvent<SVGSVGElement>) => {
         // when user holding the shift key and mis-click the background
         // preserve the current selection
+        if (mode === 'select') {
+            const { x, y } = getMousePosition(e);
+            // console.info(
+            //     selectStart,
+            //     (x * svgViewBoxZoom) / 100 + svgViewBoxMin.x,
+            //     (y * svgViewBoxZoom) / 100 + svgViewBoxMin.y
+            // );
+            HandleSelectTool(
+                graph.current,
+                selectStart.x,
+                selectStart.y,
+                (x * svgViewBoxZoom) / 100 + svgViewBoxMin.x,
+                (y * svgViewBoxZoom) / 100 + svgViewBoxMin.y
+            ).forEach(node => {
+                dispatch(addSelected(node));
+            });
+            dispatch(setMode('free'));
+            dispatch(setSelectStart({ x: 0, y: 0 }));
```

Comment:
Do we need to clear `selectStart` and `selectMoving`? `<rect />` should not be rendered as long as the mode is not `select`. Or is this used elsewhere?

Reply:
It was inited when press the board in `handleBackgroundDown`

Reply:
Yeah, I think both of them will be initialized in `handleBackgroundDown` so this line could be removed as long as we do not use them and until we reset them.

Reply:
That's right, we can delete them.

Reply:
Oh no, `selectStart` must be cleared, because the selection box will appear when I click "Select" button before actually select.

Reply:
We may clear it when choosing the select tool, but then `selectStart` still needs to be shared between components via runtime-slice. There might be a better way. Let me find it out later :)

Reply:
You're right, they can not be removed in mouse-up events. Let's keep it as it is.

Reply:
I found that `selectMoving` may be cleared as well, or the rectangle will appear once the mouse down. I think there must be something wrong, but I have limited time now.